### PR TITLE
DOCS-536: Refactor Blink an LED with Raspberry Pi tutorials

### PR DESCRIPTION
**Pull request by JessamyT**
- Combine the two tutorials into one
- Rewrite various bits, for app updates as well as for brevity etc.
  - There's still one bit (line 127) that uses "we" and "we'll"--should that come out? Wasn't sure about desired tone in this tutorial.
- Shorten and change name of file and image directories
- Update UI screenshots
- Give some images more meaningful names than "image3" etc.

### CODE REVIEW

File: docs/tutorials/get-started/blink-an-led.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+level: "Beginner"
+date: "4 October 2022"
+cost: 70
+no_list: true
+---
+
+This tutorial will show you how to use Viam to make an LED blink with a Raspberry Pi.
+This is a great place to start if you have never built a robot or a circuit before.
+
+{{<gif webm_src="/tutorials/img/blink-an-led/image9.webm" mp4_src="/tutorials/img/blink-an-led/image9.mp4" alt="A GIF of the completed project showing a blinking blue LED connected to a Raspberry Pi with jumper cables." max-width="300px">}}
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:29]: I think the video from the other tutorial is actually nicer because it's not jumpy?

File: docs/tutorials/get-started/blink-an-led.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+- [Viam Python SDK](https://python.viam.dev/) or [Viam Go SDK](https://pkg.go.dev/go.viam.com/rdk/robot/client#section-readme)
+
+## Project setup
+
+Before you build your circuit, you need to set up the operating system on your Raspberry Pi and install `viam-server` on the Pi:
+
+1. Follow the [Raspberry Pi Setup Guide](/installation/prepare/rpi-setup/) to set up the operating system.
+2. Then [install `viam-server` and connect your Pi to the Viam app](/installation/#install-viam-server).
+
+{{< snippet "social.md" >}}
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:59]: I think the snipped doesn't fit well here. 

"If you encounter issues, reach out to us on the Community Discord." would work better

File: docs/tutorials/get-started/blink-an-led.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+
+## Build the circuit
+
+The first step in this project is to build a simple LED circuit consisting of an LED and a resistor.
+The resistor protects the LED by limiting the flow of electricity through the circuit and is called a *current-limiting resistor*.
+
+Then you will make the LED circuit controllable from the Raspberry Pi by connecting the circuit to the [pins](https://en.wikipedia.org/wiki/Pin_(electronics)) on the Raspberry Pi.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:66]: It's not a circuit until connected to the PI I think because it's not, well, a connected circle, right?

File: docs/tutorials/get-started/blink-an-led.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+### Understand GPIO pinout
+
+A *general-purpose input/output* (GPIO) pin is a digital signal pin on a circuit board, like a Raspberry Pi, which may be used as an input or output, or both, and is controllable by software.
+
+<img src="../../img/blink-an-led/pi-gpio.png" alt="Photo showing a Raspberry Pi 4 with a white box around the GPIO pins on the Pi and big red letters that say, 'GPIO Pins.'" width="100%">
+
+As you may have guessed, **each pin has a specific role, and you can use it only for that role**.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:74]:
```suggestion
**Each pin has a specific role, and you can use it only for that role**.
```
The as you may have guessed isn't needed - we don't want to make them feel stupid if they didn't think that

File: docs/tutorials/get-started/blink-an-led.md [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+<img src="../../img/blink-an-led/pinout.jpg" alt="Diagram showing all of the GPIO pins on a Raspberry Pi 4 and their corresponding pin number and function." width="100%">
+
+One thing that might be confusing is the *board pin numbering* versus *GPIO pin numbering*.
+There are 40 physical pins numbered from 1 to 40.
+That is board pin numbering, corresponding to the pin's physical location on the board.
+Then there's numbering them by function or GPIO connection.
+For example, "GPIO 22".
+**When working with the GPIO pins with Viam, you will use the board pin numbers.**
+For example, board pin 38 is the same pin as GPIO pin 20, but when configuring your robot with Viam, you should refer to it as pin `38`.
+
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:88]:
```suggestion

{{< alert title="Tip" color="tip" >}}
To make it easier for you to see which pin is which, you can print out [this piece of paper at 100% scaling level](https://docs.viam.com/try-viam/rpi4_rover_leaf_A4.pdf) which has labels for the pins and carefully push it onto the pins or fold or cut it so you can hold it up to the Raspberry Pi pins. If you are having trouble punching the pins through, you can pre-punch the pin holes with a pen. Only attach the paper when the Pi is unplugged. To make attaching the paper easier, use a credit card or a small screwdriver.
{{< /alert >}}

```

File: docs/tutorials/get-started/blink-an-led.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+{{% /alert %}}
+
+Now it's time to wire your circuit according to the diagram below.
+It doesn't matter what color jumper wires you use but for reference, the blue wire below connects the LED to ground, and the red wire connects the resistor to pin 8 (GPIO 14).
+
+<img src="../../img/blink-an-led/circuit.png" alt="Circuit diagram showing a Raspberry Pi with a red connector running out of GPIO pin 8 to a 100-ohm resistor. The resistor is connected to the long lead of a red LED bulb. Finally, a blue connector connects the short lead of the LED to the ground connection on pin 6 of the Raspberry Pi GPIO pins." width="100%">
+
+The resistor and LED need to be in series as in the diagram above.
+To find the right resistor use the resistor color code -- for a 100 ohm resistor, it needs to be brown-black-brown.
+You can use your multimeter to double-check the resistor value or check yours using the photo below.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:105]:
```suggestion
You can use a multimeter to double-check the resistor value or check yours using the photo below.
```

File: docs/tutorials/get-started/blink-an-led.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+
+<img src="../../img/blink-an-led/resistor.jpg" alt="Photo of a 100-ohm resistor with text overlaid that says, in order, brown-black-brown-gold." width="50%">
+
+When hooking up the circuit, **note the *polarity* of the LED**.
+You will notice that the LED has long and short leads.
+The long lead is the positive side, which is known as the anode.
+The short lead is the negative side, which is known as the cathode.
+Connect the long anode to the resistor and the short cathode to ground (pin 6) on the Raspberry Pi.
+
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:114]: I think we're missing a step where we tell readers to turn the power back on

File: docs/tutorials/get-started/blink-an-led.md [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+## Configure your robot
+
+Before proceeding, be sure that you have [connected your Pi to the Viam app](/installation/#install-viam-server).
+
+Now it's time to configure your robot's components.
+First, go to the [Viam app](https://app.viam.com/) on your web browser and navigate to your robot's **Config** tab.
+
+<img src="../../img/blink-an-led/create-component.png" alt="Screenshot of the Viam app showing the Create component field on the Components subtab of the Config tab." width="100%">
+
+You will add a [*board component*](/components/board/) to represent your single board computer, which in this case is the Raspberry Pi.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:124]: Use imperative mood

File: docs/tutorials/get-started/blink-an-led.md [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+You'll create the new component in the **Create component** panel.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:125]: avoid future tense

File: docs/tutorials/get-started/blink-an-led.md [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+
+- Name the board whatever you like as long as you are consistent when referring to it later; we'll name it `local` since it is the board we will communicate with directly.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:127]: Do you mind adding tabs here with the config in raw json on the second tab? like on the guardian tutorial for example?

File: docs/tutorials/get-started/blink-an-led.md [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+Your board component pane will look like this:
+
+<img src="../../img/blink-an-led/board-config.png" alt="Screenshot of the Viam app showing the board configuration on the Config tab. The board is named 'local' and the attributes are shown as empty braces." width="100%">
+
+Click **Save Config** at the bottom left of the page.
+
+## Control your robot using the Viam app
+
+When you configure your board component, the Viam app generates a control panel for it.
+Click the **Control** tab to view the control panels for all your robot's components (in this case, just the board).
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:141]: add link to the control tab docs

File: docs/tutorials/get-started/blink-an-led.md [hunk 12]
```diff
@@ -0,0 +1,416 @@
...
+<img src="../../img/blink-an-led/set-pin-state.png" alt="Screenshot of the Viam app showing the board configuration on the Control tab. The 'Board Local' row is expanded, and under the 'Set' row, the pin is set to '8.' A red box is around the 'Set Pin State' field." width="100%">
+
+When you set your pin to **high** the LED should illuminate.
+You can set the pin back and forth between high and low, and you will see your LED turn on or off depending on whether you have the value set to **low** or **high**.
+
+{{<gif webm_src="/tutorials/img/blink-an-led/image3.webm" mp4_src="/tutorials/img/blink-an-led/image3.mp4" alt="A GIF of the completed project showing a blinking blue LED connected to a Raspberry Pi with jumper cables." max-width="110%">}}
+
+Congratulations!
+You have just successfully used Viam to make an LED blink with a Raspberry Pi!
+You have learned how the GPIO on a Raspberry Pi works, and how to build circuits for LED bulbs.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:161]:
```suggestion
You have learned how the GPIO on a Raspberry Pi works, and how to build circuits for LED bulbs.

To make your LED blink periodically, you need to use an SDK. 
```

Comment (npentrel) [reply docs/tutorials/get-started/blink-an-led.md:161]: ^ This got people confused on discord before so worth adding

File: docs/tutorials/get-started/blink-an-led.md [hunk 13]
```diff
@@ -0,0 +1,416 @@
...
+
+## Control your robot using the Viam SDKs
+
+Now you're ready to control the LED with code so you don't have to click buttons to turn it on and off.
+
+In this section, you will be introduced to the basics of programming hardware by using either the [Viam Python SDK](https://python.viam.dev/) (software development kit) or the [Viam Go SDK](https://pkg.go.dev/go.viam.com/rdk/robot/client#section-readme) to make your LED blink.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:167]: Instead of "will be introduced", keep it about the user and what they will actively learn (instread of passively being introduced to)

File: docs/tutorials/get-started/blink-an-led.md [hunk 14]
```diff
@@ -0,0 +1,416 @@
...
+
+## Summary
+
+In this tutorial, you learned to:
+
+- Set up a circuit
+- Use the Viam app to configure and control a robot
+- Control your robot using the Viam SDK by writing a short program in either Go or Python to make an LED on your Raspberry Pi blink on and off!
+
+Now that you have completed your first robotics project, check out some of our other [tutorials](/tutorials/) to do even more with robots.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:411]:
```suggestion
Now that you have completed this robotics project, check out some of our other [tutorials](/tutorials/).
```

**Review by npentrel — COMMENTED** [review]
Good start - it's good to see these tutorials merged, that makes so much more sense

File: docs/tutorials/get-started/blink-an-led.md (hunk 1 above)
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:29]: I kinda like the lighting etc. on this one so I left it in, yeah. And then the other ones are near the part of the tute that talks about coding them so that it's actually a set interval. Can change if you want though?

File: docs/tutorials/get-started/blink-an-led.md (hunk 4 above)
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:74]: 100%

File: docs/tutorials/get-started/blink-an-led.md (hunk 5 above)
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:88]:
```suggestion

{{< alert title="Tip" color="tip" >}}
To make it easier for you to see which pin is which, you can print out [this piece of paper at 100% scaling level](https://docs.viam.com/try-viam/rpi4_rover_leaf_A4.pdf) which has labels for the pins.
Carefully push it onto the pins or fold or cut it so you can hold it up to the Raspberry Pi pins.
If you are having trouble punching the pins through, you can pre-punch the pin holes with a pen.
Only attach the paper when the Pi is unplugged.
To make attaching the paper easier, use a credit card or a small screwdriver.
{{< /alert >}}

```

File: docs/tutorials/get-started/blink-an-led.md [hunk 15]
```diff
@@ -0,0 +1,416 @@
...
+If you successfully configured your robot and it is able to connect to the Viam app, you should see something like this printed to the terminal after running your program.
+What you see here is a list of the various {{< glossary_tooltip term_id="resource" text="resources" >}} that have been configured on your robot in the Viam app.
+
+<img src="../../img/blink-an-led/image4.png" alt="A screenshot from the Visual Studio Code command line that prints the output of print(robot.resource_names) when your Raspberry Pi has correctly connected and initialized with the Viam app. The output is an array of resources that have been pulled from the Viam app. Some of these are the Vision Service, Data Manager, and Board." width="500">
+
+### Write code to blink the LED
+
+In order to interact with the GPIO pins on our Raspberry Pi, you need to import the [board component](/components/board/) from the Viam SDK.
+
+The **Code sample** tab should have automatically added the board import for you, but it doesn't hurt to double-check.
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:222]: Yeah--I wasn't sure how to approach this because I want people to learn the need to import in case they write their own code from scratch, but should this section actually even be here? (other than the go time and fmt imports)?

File: docs/tutorials/get-started/blink-an-led.md (hunk 3 above)
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:66]: -_- okay that's fair :)

File: docs/tutorials/get-started/blink-an-led.md (hunk 10 above)
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:135]: What is the objective of that? If we strongly want people to use the config builder, why add extra to an absolute beginner tutorial?

File: docs/tutorials/get-started/blink-an-led.md [hunk 16]
```diff
@@ -0,0 +1,416 @@
...
+For this tutorial's Go code, you will also want `"fmt"` (for printing) and `"time"` (for time delays) as well as the board import.
+
+In <file>blink.go</file>, and inside the `import` block at the top of the file, add the following to the import statements without removing any of the other imports:
+
+```go {class="line-numbers linkable-line-numbers"}
+import (
+  "fmt"
+  "time"
+  "go.viam.com/rdk/components/board"
+)
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:255]: I said that because I don't want Python users to think they're missing something, or to make Go users wonder why their tab is different from the Python code.

File: docs/tutorials/get-started/blink-an-led.md [hunk 17]
```diff
@@ -0,0 +1,416 @@
...
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go <INSERT PATH TO YOUR FILE>/blink.go
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+If you successfully configured your robot and it is able to connect to the Viam app, you should see something like this printed to the terminal after running your program.
+What you see here is a list of the various {{< glossary_tooltip term_id="resource" text="resources" >}} that have been configured on your robot in the Viam app.
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:224]: Okay adding a go output but they actually do see all of this because these services are built-in. Just ran and double-checked.

File: docs/tutorials/get-started/blink-an-led.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:268]: How would you go about making the tabs more similar when the Python one is just looking and seeing that the board import is there, while the Go one requires people to add things that do not auto-import? Should I take away these tabs and make an "IMPORTANT" alert that says "Go users also need to import fmt and time. If you are using Go for this tutorial, the top of your code should include these imports:
....
"?

File: docs/tutorials/get-started/blink-an-led.md (hunk 7 above)
Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:116]: Done!

File: docs/tutorials/get-started/blink-an-led.md [hunk 18]
```diff
@@ -0,0 +1,442 @@
...
+2. Then [install `viam-server` and connect your Pi to the Viam app](/installation/#install-viam-server).
+
+If you encounter issues, reach out to us on the [Viam Community Discord](https://discord.com/invite/viam).
+
+## Build the circuit
+
+The next step is to build a simple LED circuit consisting of an LED and a resistor.
+The resistor protects the LED by limiting the flow of electricity through the circuit and is called a *current-limiting resistor*.
+
+You will make the LED controllable from the Raspberry Pi by connecting it to the [pins](https://en.wikipedia.org/wiki/Pin_(electronics)) on the Raspberry Pi.
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:60]: This sentence is still a bit awkward ( the will make controllable) - can you rephrase?
```suggestion
```

File: docs/tutorials/get-started/blink-an-led.md [hunk 19]
```diff
@@ -0,0 +1,442 @@
...
+Run the code by typing the following command into the terminal, with the correct path to your file:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+python3 <INSERT PATH TO YOUR FILE>/blink.py
+```
+
+If you successfully configured your robot and it is able to connect to the Viam app, you should see something like this printed to the terminal after running your program.
+What you see here is a list of the various {{< glossary_tooltip term_id="resource" text="resources" >}} that have been configured on your robot in the Viam app (the board, as well as various built-in services).
+
+<img src="../../img/blink-an-led/python-output.png" alt="A screenshot from the Visual Studio Code command line that prints the output of print(robot.resource_names) when your Raspberry Pi has correctly connected and initialized with the Viam app. The output is an array of resources that have been pulled from the Viam app. Some of these are the Vision Service, Data Manager, and Board." width="500">
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:214]: I think better than a screenshot here would be the actual output in code format (same for python )

File: docs/tutorials/get-started/blink-an-led.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (npentrel) [reply docs/tutorials/get-started/blink-an-led.md:268]: No. The tabs being there is good.

python:
At the top of your blink.py file, make sure it includes the following import:

go
At the top of your blink.go file, make sure it includes the following imports:
// code

The `fmt` package enables you to print and the `time` package enables you to add time delays.

File: docs/tutorials/get-started/blink-an-led.md (hunk 10 above)
Comment (npentrel) [reply docs/tutorials/get-started/blink-an-led.md:131]: I think it's nice to show the alternatives for people who don't want to use the builder UI and I would like us to add that to all tutorials. Then it would be the odd one out.

Comment (JessamyT) [reply docs/tutorials/get-started/blink-an-led.md:131]: ok

File: docs/tutorials/get-started/blink-an-led.md [hunk 20]
```diff
@@ -0,0 +1,498 @@
...
+
+If you encounter issues, reach out to us on the [Viam Community Discord](https://discord.com/invite/viam).
+
+## Build the circuit
+
+The next step is to build a simple LED circuit consisting of an LED and a resistor.
+The resistor protects the LED by limiting the flow of electricity through the circuit and is called a *current-limiting resistor*.
+
+You will connect the LED and resistor to the [pins](https://en.wikipedia.org/wiki/Pin_(electronics)) on the Raspberry Pi to complete the circuit.
+Since the only power for this circuit comes from a Pi pin, controlling the state of the pin will toggle the LED on or off.
```
Comment (JessamyT) [inline docs/tutorials/get-started/blink-an-led.md:61]: @npentrel Maybe like this? Other ideas?

Comment (npentrel) [reply docs/tutorials/get-started/blink-an-led.md:61]: Yes I like it :)

File: docs/tutorials/get-started/blink-an-led.md [hunk 21]
```diff
@@ -208,10 +235,32 @@ Run the code by typing the following command into the terminal, with the correct
 python3 <INSERT PATH TO YOUR FILE>/blink.py
 ```
 
-If you successfully configured your robot and it is able to connect to the Viam app, you should see something like this printed to the terminal after running your program.
+If you successfully configured your robot and it is able to connect to the Viam app, you should see something like the following printed to the terminal after running your program.
 What you see here is a list of the various {{< glossary_tooltip term_id="resource" text="resources" >}} that have been configured on your robot in the Viam app (the board, as well as various built-in services).
 
-<img src="../../img/blink-an-led/python-output.png" alt="A screenshot from the Visual Studio Code command line that prints the output of print(robot.resource_names) when your Raspberry Pi has correctly connected and initialized with the Viam app. The output is an array of resources that have been pulled from the Viam app. Some of these are the Vision Service, Data Manager, and Board." width="500">
+```sh {id="terminal-prompt" class="command-line" data-prompt=" "}
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:241]:
```suggestion
```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
```

File: docs/tutorials/get-started/blink-an-led.md [hunk 22]
```diff
@@ -208,10 +235,32 @@ Run the code by typing the following command into the terminal, with the correct
...
+$ python3 my-project-folder/blink.py
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:242]:
```suggestion
python3 my-project-folder/blink.py
```

File: docs/tutorials/get-started/blink-an-led.md (hunk 21 above)
Comment (npentrel) [reply docs/tutorials/get-started/blink-an-led.md:241]: the same goes for the go snippet

**Review by npentrel — COMMENTED** [review]
Lmk once the code snippets are done :)

File: docs/tutorials/get-started/blink-an-led.md [hunk 23]
```diff
@@ -0,0 +1,498 @@
...
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go run blink.go
+```
+
+If you successfully configured your robot and it is able to connect to the Viam app, you should see something like the following printed to the terminal after running your program.
+What you see here is a list of the various {{< glossary_tooltip term_id="resource" text="resources" >}} that have been configured on your robot in the Viam app (the board, as well as various built-in services).
+
+```sh {id="terminal-prompt" class="command-line" data-prompt=" "}
+$ username@hostname.local:~/my-project-folder $ go run blink.go
```
Comment (JessamyT) [inline docs/tutorials/get-started/blink-an-led.md:286]:
```suggestion
```sh {id="terminal-prompt" class="command-line" data-prompt="$" data-output="2-25"}
username@hostname.local:~/my-project-folder $ go run blink.go
```

File: docs/tutorials/get-started/blink-an-led.md [hunk 24]
```diff
@@ -0,0 +1,498 @@
...
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go run blink.go
+```
+
+If you successfully configured your robot and it is able to connect to the Viam app, you should see something like the following printed to the terminal after running your program.
+What you see here is a list of the various {{< glossary_tooltip term_id="resource" text="resources" >}} that have been configured on your robot in the Viam app (the board, as well as various built-in services).
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$" data-output="2-25"}
+username@hostname.local:~/my-project-folder $ go run blink.go
```
Comment (npentrel) [inline docs/tutorials/get-started/blink-an-led.md:286]:
```suggestion
go run blink.go
```

**Review by npentrel — APPROVED** [review]
Lgtm % final comment
